### PR TITLE
Feature/add label endpoint

### PR DESCRIPTION
**Pull request by RajanGuptaNZ**
#### Description
Added an addLabel endpoint to the backend - POST /addLabel
Body of request is:
```
{
    "projectID": x,
    "documentID": x
    "userID": x,
    "label": "xxx"
}
```

Responds with:
```
{
    "message": "Added label successfully",
    "status_code": 200
} 
``` 
or 
```
{
    "message": "Missing xxx",
    "status_code": 400
}
```

#### Review Checklist
- [x] The requirements on the description have been met
- [x] The relevant documentation has been updated:
    - [x] code comments for classes / methods / complex statements
- [x] Tests:
    - [x] Tested using Postman
- [ ] Tested by reviewer

### CODE REVIEW

File: backend/main.py [hunk 1]
```diff
@@ -9,6 +10,49 @@
 def hello():
     return "hello world"
 
+#Endpoint to allow adding of labels to a document
+@app.route('/addlabel', methods=['Post'])
+def addLabel():
+    if 'projectID' in request.json:
+        projectID = int(request.json['projectID'])
```
Comment (tnzhu) [inline backend/main.py:17]: The current ID's are automatically generated when documents are added into the collections, so we'll need to discuss if we're using those, or setting our own ID's.

Documents are the exception at the moment, the id field is "identifier"

We could also use the User's email as their ID

File: backend/main.py [hunk 2]
```diff
@@ -9,6 +10,49 @@
...
+    if 'label' in request.json:
+        label = Label(request.json['label'])
+    else:
+        response = {'status_code': 400,
+                    'message': "Missing label"}
+        response = make_response(response)
+        return response
+
+    ### Check user permisisions for project ###
+    ### Add to DB ###
```
Comment (tnzhu) [inline backend/main.py:49]: See `add_user_and_label` in backend/model/document.py

**Review by tnzhu — CHANGES_REQUESTED** [review]

File: backend/main.py (hunk 2 above)
Comment (saddboys) [reply backend/main.py]: Yeah we will likely use the mongoDB IDs for this purpose

File: backend/main.py (hunk 1 above)
Comment (saddboys) [reply backend/main.py:31]: Yeah we will likely use the mongodb IDs

Comment (RajanGuptaNZ) [reply backend/main.py:31]: Does this really affect what the endpoint keys are called?

File: backend/main.py (hunk 2 above)
Comment (RajanGuptaNZ) [reply backend/main.py]: I've kind of mocked this up in my new commits, but I can't fully implement it until I can get users & documents from the db

**Review by saddboys — APPROVED** [review]
looks fine

**Pull request merged**